packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -31,34 +32,83 @@ interface GoogleCalError extends Error {
 
 const ONE_MINUTE_MS = 60 * 1000;
 const CACHING_TIME = ONE_MINUTE_MS;
+const MS_PER_DAY = 24 * 60 * 60 * 1000;
+const ONE_MONTH_IN_MS = 30 * MS_PER_DAY;
 
 /** Expand the start date to the start of the month */
 function getTimeMin(timeMin: string) {
+  const date = new Date();
+  const dateMonth = date.getMonth();
   const dateMin = new Date(timeMin);
+  const dateMinMonth = dateMin.getMonth();
+  // If we browser the next month, return the start of the previous month to guarantee a cache hit
+  if (date.getFullYear() === dateMin.getFullYear() && dateMinMonth - dateMonth === 1) {
+    return new Date(date.getFullYear(), date.getMonth(), 1, 0, 0, 0, 0).toISOString();
+  }
   return new Date(dateMin.getFullYear(), dateMin.getMonth(), 1, 0, 0, 0, 0).toISOString();
 }
 
 /** Expand the end date to the end of the month */
 function getTimeMax(timeMax: string) {
+  const date = new Date();
+  const dateMonth = date.getMonth();
   const dateMax = new Date(timeMax);
-  return new Date(dateMax.getFullYear(), dateMax.getMonth() + 1, 0, 0, 0, 0, 0).toISOString();
+  const dateMaxMonth = dateMax.getMonth();
+  if (
+    // Normalize the next two months requests to guarantee a cache hit
+    date.getFullYear() === dateMax.getFullYear() &&
+    (dateMaxMonth - dateMonth === 1 || dateMaxMonth - dateMonth === 2)
+  ) {
+    return new Date(date.getFullYear(), date.getMonth() + 2, 1, 0, 0, 0, 0).toISOString();
+  }
+  return new Date(dateMax.getFullYear(), dateMax.getMonth(), 1, 0, 0, 0, 0).toISOString();
 }
 
-/**
- * Enable or disable the expanded cache
- * TODO: Make this configurable
- * */
-const ENABLE_EXPANDED_CACHE = true;
+function treatAsUTC(date: string | Date) {
+  const result = new Date(date);
+  result.setMinutes(result.getMinutes() - result.getTimezoneOffset());
+  return result;
+}
+
+function daysBetween(startDate: string | Date, endDate: string | Date) {
+  return (treatAsUTC(endDate).getTime() - treatAsUTC(startDate).getTime()) / MS_PER_DAY;
+}
+
+function addDays(startDate: string | Date, numberOfDays = 90) {
+  return new Date(treatAsUTC(startDate).getTime() + numberOfDays * MS_PER_DAY).toISOString();
+}
 
 /**
  * By expanding the cache to whole months, we can save round trips to the third party APIs.
  * In this case we already have the data in the database, so we can just return it.
  */
 function handleMinMax(min: string, max: string) {
-  if (!ENABLE_EXPANDED_CACHE) return { timeMin: min, timeMax: max };
-  return { timeMin: getTimeMin(min), timeMax: getTimeMax(max) };
+  const timeMin = getTimeMin(min);
+  const timeMax = getTimeMax(max);
+  /**
+   * Prevents quering more that 90 days
+   * @see https://github.com/calcom/cal.com/pull/11962
+   */
```

Comment:
@CarinaWolli please let me know it this works as well 🙏🏽

Reply:
Would fix the thrown error, but it then would not get any busy times after the 90 days